src/mixins/itext_behavior.mixin.js

```diff
@@ -412,6 +418,252 @@
+            this.hiddenTextarea && (this.hiddenTextarea.value = this.text);
+            this._updateTextarea();
+            this.fire('changed', { index: selectionStart, action: 'dragend' });
+            this.canvas.fire('text:changed', { target: this });
+            this.canvas.requestRenderAll();
+          }
+          else {
+            this.selectionStart = selectionStart;
+            this.selectionEnd = selectionEnd - trailingSpace;
+          }
+          this.exitEditing();
+          this.__lastSelected = false;
+        }
+
+        this.fire('dragend', { e: e });
+      }
+
+      this.__dragImageDisposer && this.__dragImageDisposer();
+      delete this.__dragImageDisposer;
+      delete this.__dragStartSelection;
+      this.__isDraggingOver = false;
+    },
+
+    /**
+     * support native like text dragging
+     *
+     * Override the `text/plain | application/fabric` types of {@link DragEvent#dataTransfer}
+     * in order to change the drop value or to customize styling respectively, by listening to the `drop:before` event
+     * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#performing_a_drop
+     * @private
+     */
+    dropHandler: function (options) {
+      var e = options.e;
+      this.__isDraggingOver = false;
+      // inform browser that the drop has been accepted
+      e.preventDefault();
+      var insert = e.dataTransfer.getData('text/plain');
+      if (insert) {
+        var insertAt = this.selectionStart;
+        var data = e.dataTransfer.types.includes('application/fabric') ?
+          JSON.parse(e.dataTransfer.getData('application/fabric')) :
+          {};
+        var styles = data.styles;
+        var trailingSpace = data.trailingSpace, trailing = insert[Math.max(0, insert.length - 1)];
+        this.canvas.discardActiveObject();
+        this.canvas.setActiveObject(this);
+        this.enterEditing(e);
+        var isSpace = /\s/;
+        var insertingOnSpace = isSpace.test(this._text[insertAt]);
+        var insertingAfterSpace = isSpace.test(this._text[insertAt - 1]);
+        var seletionStartOffset = 0;
+        //  drag and drop in same instance
+        if (this.__dragStartSelection) {
+          var selectionStart = this.__dragStartSelection.selectionStart;
+          var selectionEnd = this.__dragStartSelection.selectionEnd;
+          if (insertAt > selectionStart && insertAt <= selectionEnd) {
+            insertAt = selectionStart;
+            trailingSpace = false;
+          }
+          else if (insertAt > selectionEnd) {
+            insertAt -= selectionEnd - selectionStart;
+          }
+          this.insertChars('', null, selectionStart, selectionEnd);
+          // prevent `dragend` from handling event
+          delete this.__dragStartSelection;
+        }
+        //  handle spacing
+        if (trailingSpace && this.findLineBoundaryLeft(insertAt) < insertAt && !insertingAfterSpace
```

Comment:
@Lazauya Thanks! Did you live test it?
I think I encountered a bug when dragging a selected word. In some cases white space injection might be wrong.
The logic starts here